airflow-core/src/airflow/models/dagrun.py

```diff
@@ -801,6 +801,7 @@ def fetch_task_instances(
                 TI.dag_id == dag_id,
                 TI.run_id == run_id,
             )
+            .order_by(TI.task_id)
```

Comment:
```suggestion
            .order_by(TI.task_id, TI.map_index)
```

If we’re after stability. This _might_ actually be faster since we have a unique constraint on `(dag_id, task_id, run_id, map_index)` (the other two are filtered on in this query) and that’s generally backed by an index.

Reply:
Yep.